src/Desktop.Analyzers/Core/DoNotUseInsecureDTDProcessing.cs

```diff
@@ -0,0 +1,860 @@
+                        if (version != null)
+                        {
+                            context.RegisterCodeBlockStartAction<TLanguageKindEnum>(
+                                (c) =>
+                                {
+                                    RegisterAnalyzer(c, xmlTypes, version);
+                                });
+                        }
+                    }
+                });
+        }
+
+        private static bool ReferencesAnyTargetType(CompilationSecurityTypes types)
+        {
+            return types.XmlDocument != null
+                || types.XmlNode != null
+                || types.XmlReader != null
+                || types.XmlTextReader != null
+                || types.XPathDocument != null
+                || types.XmlSchema != null
+                || types.DataSet != null
+                || types.DataTable != null
+                || types.DataViewManager != null
+                || types.XmlSerializer != null;
+        }
+
+        private static DiagnosticDescriptor CreateDiagnosticDescriptor(LocalizableResourceString messageFormat, LocalizableResourceString description, string helpLink = null)
+        {
+            return new DiagnosticDescriptor(RuleId,
+                                            SecurityDiagnosticHelpers.GetLocalizableResourceString(nameof(DesktopAnalyzersResources.InsecureXmlDtdProcessing)),
+                                            messageFormat,
+                                            DiagnosticCategory.Security,
+                                            DiagnosticSeverity.Warning,
+                                            isEnabledByDefault: true,
+                                            description: description,
+                                            helpLinkUri: helpLink,
+                                            customTags: WellKnownDiagnosticTags.Telemetry);
+        }    
+
+        protected abstract void RegisterAnalyzer(CodeBlockStartAnalysisContext<TLanguageKindEnum> context, CompilationSecurityTypes types, Version targetFrameworkVersion);
+
+        protected abstract class Analyzer
+        {
+            // .NET frameworks >= 4.5.2 have secure default settings
+            private static readonly Version minSecureFxVersion = new Version(4, 5, 2);
+
+            private readonly CompilationSecurityTypes xmlTypes;
+            private readonly SyntaxNodeHelper syntaxNodeHelper;
+            private readonly bool isFrameworkSecure;
+
+            private readonly Dictionary<ISymbol, XmlDocumentEnvironment> xmlDocumentEnvironments = new Dictionary<ISymbol, XmlDocumentEnvironment>();
+            private readonly Dictionary<ISymbol, XmlTextReaderEnvironment> xmlTextReaderEnvironments = new Dictionary<ISymbol, XmlTextReaderEnvironment>();
+            private readonly Dictionary<ISymbol, XmlReaderSettingsEnvironment> xmlReaderSettingsEnvironments = new Dictionary<ISymbol, XmlReaderSettingsEnvironment>();
```

Comment:
Since PR is already out and you will fix these style problem any way, just file an issue on this and fix them together in one shot. 

From now on, I will be following this style.